mllib/src/test/scala/org/apache/spark/ml/optim/aggregator/LogisticAggregatorSuite.scala

```diff
@@ -32,21 +32,21 @@ class LogisticAggregatorSuite extends SparkFunSuite with MLlibTestSparkContext {
 
   override def beforeAll(): Unit = {
     super.beforeAll()
-    instances = Array(
+    instances = standardize(Array(
```

Comment:
Just curious, why do these have to be standardized now?
One thing I want to be careful about is not introducing bugs of course, and so having it pass existing tests as-is is a good indicator that it doesn't. This may be fine but are we changing inputs to match old outputs - did behavior change?

Reply:
Two reasons:
1, we do need to standardize the input at each iteration;
2, if we keep this, then it is hard to use matrix operations in aggregator, since we  need: 1, element division, 2, broadcasting (block.matrix / stdvec), 3, special logic that if std==0 then output=0; the first 2 point may need to use `breeze` instead of BLAS, the last point may even make `breeze` not suitable.

Reply:
This may be a dumb question, but you're not saying that LR input itself has to be standardized first, right? this is just required to meet preconditions of this aggregator? just trying to get a handle on why it wasn't required before. LR now standardizes internally?

Reply:
Yes, both LinearSVC and LR always standardize the input vector in gradient computation in aggregator.
In existing single-instance `add(instance: Instance)` method, feature value `value` is always used along with std as ` value / localFeaturesStd(index)`